lib/activerecord_hoarder/batch.rb

```diff
@@ -2,25 +2,44 @@ module ::ActiverecordHoarder
   class Batch
     RECORD_DATE_FIELD = "created_at"
 
-    def self.from_records(record_data)
-      record_data.present? ? new(record_data) : nil
-    end
-
-    def initialize(record_data)
+    def initialize(record_data, delete_transaction: nil)
       @record_data = record_data
       @serializer = ::ActiverecordHoarder::Serializer
+      @delete_transaction = delete_transaction
+    end
+
+    def content_string
+      @serializer.create_archive(@record_data)
     end
 
     def date
-      @date ||= @record_data.first[RECORD_DATE_FIELD].to_date
+      @date ||= extract_date
+    end
+
+    def delete_records!
+      raise(NameError, "batch instantiated without delete transaction") if !@delete_transaction.present?
```

Comment:
Don't believe should be NameError as caller/initializer didn't specific a name, they failed to provide an optional argument.